src/puppetlabs/puppetdb/scf/hash_debug.clj

```diff
@@ -35,63 +36,45 @@
   "Updates the edge relationships from the DB to keywords for easier diffing
    with the new catalog."
   [edges]
-  (map #(update-in % [:relationship] keyword) edges))
+  (map #(update % :relationship keyword) edges))
 
 (defn resource-params->kwds
   "Updates the resource parameter keys from the DB to keywords for easier diffing
    with the new catalog."
   [resources]
-  (map (fn [resource]
-         (update-in resource [:parameters] #(kitchensink/mapkeys keyword %)))
-       resources))
+  (map #(update % :parameters (partial kitchensink/mapkeys keyword)) resources))
 
-(defn query-resources
-  [certname
-   version]
-  (let [query-sql (resources/query->sql version ["=" "certname" certname])
-        results (resources/query-resources version query-sql "")]
-    (:result results)))
+(defn get-entity
+  "An unfortunate copy of `query_eng.clj`'s `stream-query-result` without the `with-transacted-connection`
+  because we use the old version of jdbc which means that we don't pass the db-spec down through
+  this codepath and need to rely on the dynamic db connection."
+  [entity version query]
+  (let [[query->sql munge-fn] (eng/entity->sql-fns entity version {} "")
+        {:keys [results-query]} (query->sql query)]
+    (jdbc/with-query-results-cursor results-query (comp doall munge-fn))))
 
-(defn query-edges
-  [certname
-   version]
-  (let [query-sql (edges/query->sql version ["=" "certname" certname])
-        results (edges/query-edges version query-sql "")]
-    (:result results)))
-
-(defn expand-catalog-edges
-  [edges
-   certname
-   version]
-  (if (:data edges)
-    edges
-    (assoc edges :data (query-edges certname version))))
-
-(defn expand-catalog-resources
-  [resources
-   certname
-   version]
-  (if (:data resources)
-    resources
-    (assoc resources :data (query-resources certname version))))
-
-(defn query-catalog
-  [certname
-   version]
-  (-> (qcat/status version certname "")
-      (update-in [:resources] expand-catalog-resources certname version)
-      (update-in [:edges] expand-catalog-edges certname version)))
+(defn maybe-expand
+  [parent child version query]
+  (if (contains? (get parent child) :data)
+    parent
+    (assoc-in parent [child :data] (get-entity child version query))))
 
 (defn diffable-old-catalog
   "Query for the existing catalog from the DB, prep the results
    for easy diffing."
   [certname]
-  (-> (query-catalog certname :v4)
-      catalogs/catalog-query->wire-v6
-      (update-in [:edges] edge-relationships->kwds)
-      (update-in [:resources] resource-params->kwds)))
+  (let [version :v4
```

Comment:
Yeah I thought about fixing that here, the other issue i'd like to fix is passing down the db-spec so that we don't need to duplicate the stream-query-results logic, but didn't want to make things too big